pymtl/parser/bludatalog.py

```
@@ -479,41 +475,21 @@ def __getFILRow(line, keyMap):
         row['Channel'] = int(line[keyMap['INDEX_FIL_CHANNEL']])
     except (ValueError, IndexError, KeyError):
         row['Channel'] = None
-    if keyMap['VERSION'] == 5:
-        try:
-            row['Duration'] = float(line[keyMap['INDEX_FIL_DUR']+keyMap['INDEX_FIL_SPECTRUM']+int(line[keyMap['INDEX_FIL_SPECLENGTH']])])
-        except (ValueError, IndexError, KeyError):
-            row['Duration'] = None
-        try:
-            row['Valid'] = line[keyMap['INDEX_FIL_VALID']]
-        except (ValueError, IndexError, KeyError):
-            row['Valid'] = None
-        try:
-            spec = []
-            for i in range(keyMap['INDEX_FIL_SPECTRUM'],keyMap['INDEX_FIL_SPECTRUM']+int(line[keyMap['INDEX_FIL_SPECLENGTH']])):
-                try:
-                    spec.append(float(line[i]))
-                except (ValueError, IndexError, KeyError):
-                    spec.append(None)
-            row['Spectrum'] = spec
-        except (ValueError, IndexError, KeyError):

```

Created on: [2018-12-14T18:55:27Z], Created by: [glancaster]
So we quietly do what with the exception?

---

Created on: [2018-12-14T19:35:50Z], Created by: [jturner]
The reason we have these exception handling statements in all our get__Csv functions is because we're building up rows to add to an output .csv file. We support translating older BLU log files (v4) to v5, and v5 introduces new fields that weren't in the v4 log. These functions try to populate as many of the v5 fields as we can from the dictionary we've built up from the input log. Any field we didn't find will not have a key, creating an exception. The handling is to add a default placeholder value, to preserve the structure of the log entry.

We're not printing errors because we expect this to happen. Perhaps we could change all the 0s to N/A or -1 values, but this could cause problems for other parsers trying to read the values, and not getting data types they're expecting.

It might be more clear to change the exception types so that they only catch KeyErrors, to communicate that failing to find the given key is what we're handling here.

---

Created on: [2018-12-14T19:52:43Z], Created by: [glancaster]
This information needs to be in comments in the code so that poor soles like me understand why it's okay to quietly ignore exceptions.